parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -676,38 +680,21 @@ loop = do
       -- > links List.map (.Docs .English)
       -- > links List.map -- give me all the
       -- > links Optional License
-      LinksI src mdTypeStr -> do
-        let srcle = toList (getHQ'Terms src) -- list of matching src terms
-            srclt = toList (getHQ'Types src) -- list of matching src types
-            p = resolveSplit' src -- ex: the parent of `List.map` - `List`
-            mdTerms = foldl' Metadata.merge mempty [
-              BranchUtil.getTermMetadataUnder p r root0 | r <- srcle ]
-            mdTypes = foldl' Metadata.merge mempty [
-              BranchUtil.getTypeMetadataUnder p r root0 | r <- srclt ]
-            allMd = Metadata.merge mdTerms mdTypes
-        selection <- case mdTypeStr of
-          Nothing -> pure $ Right (Map.keysSet allMd)
-          Just mdTypeStr -> parseType input mdTypeStr <&> \case
-            Left e -> Left e
-            Right typ -> Right (Set.singleton (Type.toReference typ))
-        case selection of
-          Left e -> respond e
-          Right selection -> do
-            let allMd' = Map.restrictKeys allMd selection
-                allRefs = toList (Set.unions (Map.elems allMd'))
-            termDisplays <- Map.fromList <$> do
-              terms <- filterM (eval . IsTerm) allRefs
-              traverse (\r -> (r,) <$> loadTermDisplayThing r) terms
-            --   typeDisplays <- Map.fromList <$> do
-            --     types <- filterM (eval . IsType) allRefs
-            --     traverse (\r -> (r,) <$> loadTypeDisplayThing r) types
-            ppe <- prettyPrintEnv =<< makePrintNamesFromLabeled' (deps termDisplays)
-            respond $ DisplayLinks ppe allMd' mempty termDisplays
-          where
-            deps :: Map Reference (DisplayThing (Term v Ann)) -> Set LabeledDependency
-            deps dts = Set.map LD.termRef (Map.keysSet dts)
-                    <> foldMap Term.labeledDependencies
-                               (mapMaybe DT.toMaybe $ Map.elems dts)
+      LinksI src mdTypeStr -> getLinks input src (Right mdTypeStr) >>= \case
+        Left e -> respond e
+        Right (ppe, out) -> do
+          numberedArgs .= fmap (HQ.toString . view _1) out
+          respond $ ListOfLinks ppe out
```

Comment:
Note - moved bulk of `LinksI` implementation out into helper function so it could be used by both `docs` command and `links` commands.